app/helpers/practices_helper.rb

```diff
@@ -148,4 +148,57 @@ def get_tool_resource_text(resource_type)
     }
     content[resource_type.to_sym]
   end
+
+  def update_category_usages
+    s_query = params["query"]
+    cat_rec = Category.where("name ILIKE ?", s_query.downcase).first
+    return if cat_rec.blank?
+    cat_id = cat_rec.id
+    last_ahoy_event = AhoyEvent.where(name: 'Category selected').last(1)
+    if last_ahoy_event.blank?
+      store_chosen_categories(s_query, nil) unless s_query.blank?
+    else
+      event_tm = last_ahoy_event[0].time
+      ev_user_id = last_ahoy_event[0].user_id
+      last_ev_cat_id = last_ahoy_event[0].properties["category_id"]
+      same_cat_id = cat_id == last_ev_cat_id
+      same_visit = (DateTime.now.to_time.utc - event_tm) <= 2
+      same_user = current_user.id == ev_user_id
```

Comment:
good catch... I'll just return from this method if current_user.blank?

Reply:
added a check for current_user before setting this.  I think this will work... 
same_user = current_user.present? ? current_user.id == ev_user_id : ev_user_id.blank?
if current_user is null and previous user that committed is null then we would return true.. if pevious user is not null we would return false.  Let me know if you agree.  Thx